pkg/engine/executor/executor.go

```diff
@@ -5,17 +5,21 @@ import (
 	"errors"
 	"fmt"
 
+	"github.com/grafana/loki/v3/pkg/dataobj"
 	"github.com/grafana/loki/v3/pkg/engine/planner/physical"
+	"github.com/thanos-io/objstore"
 )
 
 type Config struct {
 	BatchSize int64 `yaml:"batch_size"`
+	Bucket    objstore.Bucket
```

Comment:
I threw this in as a hack but I know it's not the way we want to do this.

How should the objstore bucket be passed in?